python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -97,6 +117,26 @@ def _get_planned_materializations_for_run(self, run_id: str) -> AbstractSet[Asse
         ).records
         return set(cast(AssetKey, record.asset_key) for record in materializations_planned)
 
+    @cached_method
+    def _get_expected_materializations_for_run(self, run_id: str) -> AbstractSet[AssetKey]:
```

Comment:
This makes me think "what's the difference between a 'planned materialization' and an 'expected materialization'?" What do you think about calling these something like `_get_assets_with_materialization_planned_events_for_run` and `_get_planned_materializations_for_run`?

Alternatively, we could just not make it so complicated (my fault) and always just look at the materialization_planned events.

Reply:
mm yeah good point -- I do think having the code path branch here is a good idea because a large percentage of jobs will go down the cheap path (basically any ad-hoc run and any run kicked off by this sensor), so skipping the expensive query as often as possible is nice

will rename though